### PR TITLE
Check model access

### PR DESCRIPTION
**Pull request by zikolach**
Based on #164 (to be merged first)

Add core API method to check model access using auth server REST API.

PR also replaces and add a number of new java dependencies (such as jersey http client, jackson json parser etc) additionally upgrading jersey version to overcome incompatibilities with java 1.9+ versions.

**PR checklist:**

- [x] Tests added.
- [x] Documentation added.
- [ ] Release notes updated. Add a single line at the top of `RELEASE_NOTES.md` similar to…
  ```
  - [#999](https://github.com/iiasa/ixmp/pull/999) Description from above.
  ```
  …then, delete this sentence to confirm you've understood the instructions.

### CODE REVIEW

File: ixmp/core.py [hunk 1]
```diff
@@ -312,6 +314,25 @@ def add_region_synomym(self, region, mapped_to):
...
+        if isinstance(models, str):
+            return self._jobj.checkModelAccess(user, access, models)
+        else:
+            models_list = java.LinkedList()
+            for model in models:
+                models_list.add(model)
+            access_map = self._jobj.checkModelAccess(user, access, models_list)
+            result = {}
+            for model in models:
+                result[model] = access_map.get(model) == 1
```
[the remark this answers is not in the document]
Comment (zikolach) [reply ixmp/core.py]: Thanks for suggestion! What about conversion of jpype's "boolean" (1/0) to python one?

File: ixmp/core.py [hunk 2]
```diff
@@ -312,6 +314,25 @@ def add_region_synomym(self, region, mapped_to):
         else:
             _logger_region_exists(_regions, region)
 
+    def check_access(self, user, models, access="view"):
```
Comment (danielhuppmann) [inline ixmp/core.py]: please use single hyphens `'view'`  for consistency

File: ixmp/core.py [hunk 3]
```diff
@@ -439,6 +460,12 @@ def version(self):
 
     # functions for importing and retrieving timeseries data
 
+    def preload_timeseries(self):
+        """Preload Timeseries data to in-memory cache. Useful for bulk updates.
```
Comment (danielhuppmann) [inline ixmp/core.py]: please write timeseries with lowercase

File: ixmp/core.py [hunk 4]
```diff
@@ -439,6 +460,12 @@ def version(self):
...
+        """
+
```
Comment (danielhuppmann) [inline ixmp/core.py]: no empty line, please

File: ixmp/core.py [hunk 5]
```diff
@@ -312,6 +314,25 @@ def add_region_synomym(self, region, mapped_to):
         else:
             _logger_region_exists(_regions, region)
 
+    def check_access(self, user, models, access="view"):
+        """Check access to specific model
+
+        :param user: registered user name
```
Comment (danielhuppmann) [inline ixmp/core.py]: please follow the docstrings convention used for other functions

```
Parameters
----------
user: str
    user name
```

File: ixmp/core.py (hunk 4 above)
Comment (zikolach) [reply ixmp/core.py]: done

File: ixmp/core.py (hunk 5 above)
Comment (zikolach) [reply ixmp/core.py]: done

File: ixmp/core.py (hunk 3 above)
Comment (zikolach) [reply ixmp/core.py]: done

File: ixmp/core.py (hunk 2 above)
Comment (zikolach) [reply ixmp/core.py]: done

**Comment by zikolach** [comment]
Will rebase this PR

**Comment by zikolach** [comment]
PR was rebased onto current master, Please re-review

**Comment by zikolach** [comment]
Thanks @khaeru! Tests passed

**Pull request merged**